ProcessLib/ThermoHydroMechanics/ThermoHydroMechanicsFEM-impl.h

```diff
@@ -0,0 +1,514 @@
+                                      displacement_size);
+
+    auto local_Jac = MathLib::createZeroedMatrix<
+        typename ShapeMatricesTypeDisplacement::template MatrixType<
+            temperature_size + displacement_size + pressure_size,
+            temperature_size + displacement_size + pressure_size>>(
+        local_Jac_data, displacement_size + pressure_size + temperature_size,
+        displacement_size + pressure_size + temperature_size);
+
+    auto local_rhs = MathLib::createZeroedVector<
+        typename ShapeMatricesTypeDisplacement::template VectorType<
+            displacement_size + pressure_size + temperature_size>>(
+        local_rhs_data, displacement_size + pressure_size + temperature_size);
+
+    typename ShapeMatricesTypePressure::NodalMatrixType MTT;
+    MTT.setZero(temperature_size, temperature_size);
+
+    typename ShapeMatricesTypePressure::NodalMatrixType KTT_coeff;
+    KTT_coeff.setZero(temperature_size, temperature_size);
+
+    typename ShapeMatricesTypePressure::NodalMatrixType KTT;
+    KTT.setZero(temperature_size, temperature_size);
+
+    typename ShapeMatricesTypePressure::NodalMatrixType KTp;
+    KTp.setZero(temperature_size, pressure_size);
+
+    typename ShapeMatricesTypePressure::NodalMatrixType KTp_coeff;
+    KTp_coeff.setZero(temperature_size, pressure_size);
+
+    typename ShapeMatricesTypePressure::NodalMatrixType laplace_p;
+    laplace_p.setZero(pressure_size, pressure_size);
+
+    typename ShapeMatricesTypePressure::NodalMatrixType storage_p;
+    storage_p.setZero(pressure_size, pressure_size);
+
+    typename ShapeMatricesTypePressure::NodalMatrixType storage_T;
+    storage_T.setZero(pressure_size, temperature_size);
+
+    typename ShapeMatricesTypeDisplacement::template MatrixType<
+        displacement_size, pressure_size>
+        Kup;
+    Kup.setZero(displacement_size, pressure_size);
+    typename ShapeMatricesTypeDisplacement::template MatrixType<
+        displacement_size, temperature_size>
+        KuT;
+    KuT.setZero(displacement_size, temperature_size);
+
+    double const& dt = _process_data.dt;
+
+    SpatialPosition x_position;
+    x_position.setElementID(_element.getID());
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        x_position.setIntegrationPoint(ip);
+        auto const& w = _ip_data[ip].integration_weight;
+
+        auto const& N_u_op = _ip_data[ip].N_u_op;
+
+        auto const& N_u = _ip_data[ip].N_u;
+        auto const& dNdx_u = _ip_data[ip].dNdx_u;
+
+        auto const& N_p = _ip_data[ip].N_p;
+        auto const& dNdx_p = _ip_data[ip].dNdx_p;
+
+        // same shape function for pressure and temperature since they have the
+        // same order
+        auto const& N_T = N_p;
+        auto const& dNdx_T = dNdx_p;
+        auto const T_int_pt = N_T * T;
+        // auto const p_int_pt = N_T * p;
```

Comment:
Why is this comment so important to leave it in the code?

Reply:
This comment is removed !